### PR TITLE
[lfreist-hwinfo] Add new port

### PR DESCRIPTION
**Pull request by Hlongyu**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by WangWeiLin-MV** [comment]
Wait for #40708

File: ports/lfreist-hwinfo/usage
Comment (WangWeiLin-MV) [inline ports/lfreist-hwinfo/usage:1]: Please change the upstream export config name to `lfreist-hwinfo-config`, or append `PACKAGE_NAME` to `vcpkg_cmake_config_fixup`.

File: ports/lfreist-hwinfo/portfile.cmake
Comment (WangWeiLin-MV) [inline ports/lfreist-hwinfo/portfile.cmake:1]: As a new port, please use `LF` instead of `CRLF` as the `EOL`.

File: ports/lfreist-hwinfo/usage
Comment (Hlongyu) [reply ports/lfreist-hwinfo/usage:1]: vcpkg_cmake_config_fixup seems won't change hwinfoConfig.cmake to lfreist-hwinfoConfig.cmake. I need time to make find_package(lfreist-hwinfo) works

File: ports/lfreist-hwinfo/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    REF 46690dd36727b868c5bb7a7316bb2ee52a898349
+    SHA512 31ac0f2c405a817893146f4f8899ea05d831393bd1776c12257675385be25990cb77251d644fefb8ea0d179940ff782ede9036ea42a3f3050c36a66d978f0da6
+    HEAD_REF main
+)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" HWINFO_BUILD_SHARED)
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" HWINFO_BUILD_STATIC)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (WangWeiLin-MV) [inline ports/lfreist-hwinfo/portfile.cmake:13]: All paths need to be quoted.

```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/lfreist-hwinfo/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(
+    PACKAGE_NAME lfreist-hwinfo
+    CONFIG_PATH "lib/cmake/hwinfo"
+)
+vcpkg_copy_pdbs()
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include )
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share )
```
Comment (WangWeiLin-MV) [inline ports/lfreist-hwinfo/portfile.cmake:29]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
```

File: ports/lfreist-hwinfo/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+
```
Comment (WangWeiLin-MV) [inline ports/lfreist-hwinfo/portfile.cmake:32]: Just one `LF`.

```suggestion
```

File: ports/lfreist-hwinfo/usage [hunk 4]
```diff
@@ -0,0 +1,4 @@
+lfreist-hwinfo provides CMake targets :
+    find_package(lfreist-hwinfo CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE lfreist-hwinfo::hwinfo)
+
```
Comment (WangWeiLin-MV) [inline ports/lfreist-hwinfo/usage:4]:
```suggestion
lfreist-hwinfo provides CMake targets :

    find_package(lfreist-hwinfo CONFIG REQUIRED)
    target_link_libraries(main PRIVATE lfreist-hwinfo::hwinfo)
```

**Review by WangWeiLin-MV — APPROVED** [review]
The port usage tests pass with the following triplets:

* x64-linux
* x64-windows
* x86-windows-static

**Comment by vicroms** [comment]
Thanks for resubmitting!

**Pull request merged**